rskj-core/src/test/java/co/rsk/mine/MinerUtilsTest.java

```diff
@@ -288,4 +293,30 @@ public void getAllTransactionsCheckOrderTest() {
         Assert.assertEquals(res.get(7).getGasPrice(), Coin.valueOf(100));
         Assert.assertEquals(res.get(8).getGasPrice(), Coin.valueOf(1));
     }
+
+    @Test
+    public void filterTransactions_getNonceThrowsNodeReferenceException_throwsRuntimeException() {
+        // Given
+        RskAddress senderAddressMock = new RskAddress(new byte[20]);
+        Transaction txMock = Tx.create(config, 0, 50000, 5, 0, 0, 0);
+        Mockito.when(txMock.getSender()).thenReturn(senderAddressMock);
+
+        List<Transaction> txList = new LinkedList<>();
+        txList.add(txMock);
+
+        Repository repositoryMock = Mockito.mock(Repository.class);
+        Mockito.doThrow(new NodeReferenceException("")).when(repositoryMock).getNonce(senderAddressMock);
+
+        // Then
+        try {
+            new MinerUtils().filterTransactions(new LinkedList<>(), txList, new HashMap<>(), repositoryMock, Coin.valueOf(1L));
```

Comment:
maybe I'd move `Coin.valueOf(1L);` to a class constant, since I've seen it used multiple times in this class
(just a suggestion, maybe it doesn't add value)